torch/utils/data/dataset.py

```diff
@@ -293,19 +295,42 @@ def __len__(self):
         return len(self.indices)
 
 
-def random_split(dataset: Dataset[T], lengths: Sequence[int],
+def random_split(dataset: Dataset[T], lengths: Sequence[Union[int, float]],
                  generator: Optional[Generator] = default_generator) -> List[Subset[T]]:
     r"""
     Randomly split a dataset into non-overlapping new datasets of given lengths.
+    If a list of fractions is given, the lengths will be computed automatically as
+    floor(frac * len(dataset)).
+    If the sum of the list of the fractions is not equal to 1 or the split leaves a
+    remainder,
+    a new entry will be appended
+    to the end of the list with the remaining fraction
+
     Optionally fix the generator for reproducible results, e.g.:
 
     >>> random_split(range(10), [3, 7], generator=torch.Generator().manual_seed(42))
+    >>> random_split(range(30), [0.3, 0.3, 0.4], generator=torch.Generator(
+    ).manual_seed(42))
 
     Args:
         dataset (Dataset): Dataset to be split
-        lengths (sequence): lengths of splits to be produced
+        lengths (sequence): lengths or fractions of splits to be produced
         generator (Generator): Generator used for the random permutation.
     """
+    if math.isclose(sum(lengths), 1):
+        lengths = []
+        for i, frac in enumerate(lengths):
+            if frac < 0 or frac > 1:
+                raise ValueError(f"Fraction at index {i} is not between 0 and 1")
+            n_items_in_split = int(
+                math.floor(len(dataset) * frac)  # type: ignore[arg-type]
+            )
+            lengths.append(n_items_in_split)
+        remainder = len(dataset) - sum(lengths)  # type: ignore[arg-type]
```

Comment:
Like I said, if `sum` is larger than 1. `remainder` can be negative.

Reply:
Taken care off in latest commit